src/Database/LSMTree/Internal/RunAcc.hs

```diff
@@ -71,13 +74,24 @@ data RunAcc s = RunAcc {
     , entryCount :: !(PrimVar s Int)
     }
 
+-- | See 'Database.LSMTree.Internal.BloomFilterAlloc'
+data RunBloomFilterAlloc =
+    RunAllocFixed Word64
+  | RunAllocRequestFPR Double
+  deriving stock (Show, Eq)
+
 -- | @'new' nentries@ starts an incremental run construction.
 --
 -- @nentries@ should be an upper bound on the expected number of entries in the
 -- output run.
-new :: NumEntries -> ST s (RunAcc s)
-new (NumEntries nentries) = do
-    mbloom <- Bloom.Easy.easyNew 0.02 nentries -- TODO(optimise): tune bloom filter
+new :: NumEntries -> RunBloomFilterAlloc -> ST s (RunAcc s)
+new (NumEntries nentries) alloc = do
+    mbloom <- case alloc of
+      RunAllocFixed bitsPerEntry    ->
+        let !nbits = bitsPerEntry * fromIntegral nentries
+        in  MBloom.new (monkeyHashFuncs nbits nentries) nbits
```

Comment:
Just to be sure, despite the naming, this function is not specific to the monkey calculations, right? It's just the optimal number of hash functions for a given number of bits and entries.

Reply:
Yup, the name currently only refers to `monkey` because that is where I got the equation from